### PR TITLE
Add int64 support

### PR DESCRIPTION
**Pull request by Kubuxu**

### CODE REVIEW

**Comment by Kubuxu** [comment]
cc @anorth

File: gen.go [hunk 1]
```diff
@@ -552,6 +593,35 @@ func emitCborUnmarshalStructField(w io.Writer, f Field) error {
 	}
 }
 
+var decodeInt = `
+	var extraI int64
```
Comment (whyrusleeping) [inline gen.go]: this needs to be scoped, otherwise structs with multiple int64s will bork

File: testing/types.go [hunk 2]
```diff
@@ -1,16 +1,22 @@
 package testing
 
+type SignedArray struct {
+	Signed []uint64
```
Comment (anorth) [inline testing/types.go]: These names are very confusing given the use of an unsigned type. I don't know what this is intending.

File: gen.go (hunk 1 above)
Comment (Kubuxu) [reply gen.go]: It is scoped, line 588

File: testing/types.go (hunk 2 above)
Comment (Kubuxu) [reply testing/types.go]: I was testing, it should be `int64`.

**Pull request merged**